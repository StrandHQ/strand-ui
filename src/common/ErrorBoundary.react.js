import {Component} from 'react';
import PropTypes from 'prop-types';

class ErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = {error: null};
  }

  componentDidCatch(error, errorInfo) {
    if (Raven.isSetup()) {
      this.setState({error});
      Raven.captureException(error, {extra: errorInfo});
    }
  }

  render() {
    // TODO could add a nice "there was an error, help us out" kind of screen here
    return this.props.children;
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.object,
};

export default ErrorBoundary;
