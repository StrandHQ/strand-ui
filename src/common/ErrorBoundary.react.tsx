import {Component} from 'react';
import * as Raven from 'raven-js';

interface PropTypes {
  children: object,
}

class ErrorBoundary extends Component<PropTypes> {
  constructor(props) {
    super(props);
    this.state = {error: null};
  }

  componentDidCatch(error, errorInfo) {
    if (Raven.isSetup()) {
      this.setState({error});
      Raven.captureException(error, {extra: errorInfo});
    }
  }

  render() {
    // TODO could add a nice "there was an error, help us out" kind of screen here
    return this.props.children;
  }
}

export default ErrorBoundary;
