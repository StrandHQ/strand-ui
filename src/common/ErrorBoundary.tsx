import {Component} from 'react';
import * as Raven from 'raven-js';

interface PropTypes {
  children: object,
}

class ErrorBoundary extends Component<PropTypes> {
  constructor(props: PropTypes) {
    super(props);
    this.state = {error: null};
  }

  componentDidCatch(error: Error, errorInfo: object) {
    if (Raven.isSetup()) {
      this.setState({error});
      Raven.captureException(error, {extra: errorInfo});
    }
  }

  render() {
    // TODO could add a nice "there was an error, help us out" kind of screen here
    return this.props.children;
  }
}

export default ErrorBoundary;
