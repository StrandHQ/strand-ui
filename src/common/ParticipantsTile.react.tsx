import * as React from 'react';
import {Component} from 'react';
import Divider from 'material-ui/Divider';
import List from 'material-ui/List';
import UserListItem from './UserListItem.react';
import {filter} from 'lodash';
import {GetTopicUserFragment} from '../../schema/graphql-types';

interface PropTypes {
  originalPoster: GetTopicUserFragment
  // TODO [UI-50] No nullable arrays
  participants: (GetTopicUserFragment | null)[] | null
}

class ParticipantsTile extends Component<PropTypes> {
  nonOriginalPosterParticipants(participants: (GetTopicUserFragment | null)[]) {
    return filter(participants, participant => this.props.originalPoster.id !== participant!.id);
  }

  render() {
    const participants = this.props.participants || [];
    const otherParticipants = this.nonOriginalPosterParticipants(participants);
    return (
      <List dense>
        <UserListItem user={this.props.originalPoster} isOriginalPoster/>
        <Divider inset/>
        {otherParticipants.map(participant => (
          <div key={`participant_${participant!.id}`}>
            <Divider inset/>
            <UserListItem user={participant!}/>
          </div>
        ))}
      </List>
    );
  }
}

export default ParticipantsTile;
