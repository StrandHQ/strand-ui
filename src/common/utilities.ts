export const filterFalsey = <T extends any>(values: (T | null | undefined)[]): T[] => {
  return values.filter(x => x) as T[];
};
