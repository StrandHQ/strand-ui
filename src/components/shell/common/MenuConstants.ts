interface NavigationPaths {
  [label: string] : string,
}

export const navigationLabelToPath: NavigationPaths = {
  Install: '/install',
  Strands: '/strands',
};

export default {navigationLabelToPath};
