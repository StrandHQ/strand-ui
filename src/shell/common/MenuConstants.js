export const navigationLabelToPath = {
  'Install': '/install',
};

export default {navigationLabelToPath};
