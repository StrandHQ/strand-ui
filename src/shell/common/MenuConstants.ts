interface NavigationPaths {
  [label: string] : string,
}

export const navigationLabelToPath: NavigationPaths = {
  Install: '/install',
  Topics: '/topics',
};

export default {navigationLabelToPath};
