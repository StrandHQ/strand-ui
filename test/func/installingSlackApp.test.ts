import {navigationLabelToPath} from 'src/components/shell/common/MenuConstants';
import {mountApplication} from 'test/helper/applicationMock';
import InstallationStatus from 'src/components/install/InstallationStatus';
import {flushPromises} from '../helper/utilities';
import TestStrandSlackClient from '../clients/TestStrandSlackClient';

describe('installing slack app', () => {
  it('sends the user to Slack when the user navigates to the page and clicks the button', () => {
    const graphQLMocks = {
      Query: () => ({
        me: () => null, // not logged in
      })
    };
    const wrapper = mountApplication('/', {graphQLMocks});

    // Navigate to install page
    wrapper.find(`Button[id="${navigationLabelToPath.Slack}-button"]`).simulate('click');

    // Assert that the install button goes to Slack
    const anchorHref = wrapper.find('a[id="add-to-slack-button"]').prop('href');
    expect(anchorHref).toContain('slack.com/oauth');
    expect(anchorHref).toContain(process.env.SLACK_CLIENT_ID);
  });

  it('shows a success message when a user successfully installs the app', async () => {
    const strandSlackClient = global.strandSlackClient as TestStrandSlackClient;
    const mockSlackCode = '12345';
    strandSlackClient.addValidCode(mockSlackCode);

    const graphQLMocks = {
      Query: () => ({
        me: () => null, // not logged in
      })
    };
    const wrapper = mountApplication(`${navigationLabelToPath.Slack}?code=${mockSlackCode}`, {graphQLMocks});
    await flushPromises(); // wait for installation call to resolve
    wrapper.update();

    const installationStatusComponent = wrapper.find(InstallationStatus);
    expect(installationStatusComponent).toHaveLength(1);
    expect(installationStatusComponent).toMatchSnapshot();
  });

  it('shows a failure message when we fail to install the app', async () => {
    const graphQLMocks = {
      Query: () => ({
        me: () => null, // not logged in
      })
    };
    const wrapper = mountApplication(`${navigationLabelToPath.Slack}?code=12345`, {graphQLMocks});
    await flushPromises(); // wait for GraphQL call to complete
    wrapper.update();

    const installationStatusComponent = wrapper.find(InstallationStatus);
    expect(installationStatusComponent).toHaveLength(1);
    expect(installationStatusComponent).toMatchSnapshot();
  })
});
