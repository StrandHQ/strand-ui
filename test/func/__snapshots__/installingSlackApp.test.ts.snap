// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`installing slack app shows a failure message when we fail to install the app 1`] = `
<InstallationStatus
  error="ERROR OCCURRED"
  installingSlackApplication={false}
  successInstallationSlackApplication={false}
>
  <WithStyles(Typography)
    style={
      Object {
        "color": "red",
      }
    }
    variant="subheading"
  >
    <Typography
      align="inherit"
      classes={
        Object {
          "alignCenter": "MuiTypography-alignCenter-173",
          "alignJustify": "MuiTypography-alignJustify-175",
          "alignLeft": "MuiTypography-alignLeft-172",
          "alignRight": "MuiTypography-alignRight-174",
          "body1": "MuiTypography-body1-169",
          "body2": "MuiTypography-body2-168",
          "button": "MuiTypography-button-171",
          "caption": "MuiTypography-caption-170",
          "colorError": "MuiTypography-colorError-183",
          "colorInherit": "MuiTypography-colorInherit-179",
          "colorPrimary": "MuiTypography-colorPrimary-180",
          "colorSecondary": "MuiTypography-colorSecondary-181",
          "colorTextSecondary": "MuiTypography-colorTextSecondary-182",
          "display1": "MuiTypography-display1-164",
          "display2": "MuiTypography-display2-163",
          "display3": "MuiTypography-display3-162",
          "display4": "MuiTypography-display4-161",
          "gutterBottom": "MuiTypography-gutterBottom-177",
          "headline": "MuiTypography-headline-165",
          "noWrap": "MuiTypography-noWrap-176",
          "paragraph": "MuiTypography-paragraph-178",
          "root": "MuiTypography-root-160",
          "subheading": "MuiTypography-subheading-167",
          "title": "MuiTypography-title-166",
        }
      }
      color="default"
      gutterBottom={false}
      headlineMapping={
        Object {
          "body1": "p",
          "body2": "aside",
          "display1": "h1",
          "display2": "h1",
          "display3": "h1",
          "display4": "h1",
          "headline": "h1",
          "subheading": "h3",
          "title": "h2",
        }
      }
      noWrap={false}
      paragraph={false}
      style={
        Object {
          "color": "red",
        }
      }
      variant="subheading"
    >
      <h3
        className="MuiTypography-root-160 MuiTypography-subheading-167"
        style={
          Object {
            "color": "red",
          }
        }
      >
        Failed to install.
        <br />
        Error: 
        ERROR OCCURRED
      </h3>
    </Typography>
  </WithStyles(Typography)>
</InstallationStatus>
`;

exports[`installing slack app shows a success message when a user successfully installs the app 1`] = `
<InstallationStatus
  installingSlackApplication={false}
  successInstallationSlackApplication={true}
>
  <WithStyles(Typography)
    style={
      Object {
        "color": "green",
      }
    }
    variant="subheading"
  >
    <Typography
      align="inherit"
      classes={
        Object {
          "alignCenter": "MuiTypography-alignCenter-173",
          "alignJustify": "MuiTypography-alignJustify-175",
          "alignLeft": "MuiTypography-alignLeft-172",
          "alignRight": "MuiTypography-alignRight-174",
          "body1": "MuiTypography-body1-169",
          "body2": "MuiTypography-body2-168",
          "button": "MuiTypography-button-171",
          "caption": "MuiTypography-caption-170",
          "colorError": "MuiTypography-colorError-183",
          "colorInherit": "MuiTypography-colorInherit-179",
          "colorPrimary": "MuiTypography-colorPrimary-180",
          "colorSecondary": "MuiTypography-colorSecondary-181",
          "colorTextSecondary": "MuiTypography-colorTextSecondary-182",
          "display1": "MuiTypography-display1-164",
          "display2": "MuiTypography-display2-163",
          "display3": "MuiTypography-display3-162",
          "display4": "MuiTypography-display4-161",
          "gutterBottom": "MuiTypography-gutterBottom-177",
          "headline": "MuiTypography-headline-165",
          "noWrap": "MuiTypography-noWrap-176",
          "paragraph": "MuiTypography-paragraph-178",
          "root": "MuiTypography-root-160",
          "subheading": "MuiTypography-subheading-167",
          "title": "MuiTypography-title-166",
        }
      }
      color="default"
      gutterBottom={false}
      headlineMapping={
        Object {
          "body1": "p",
          "body2": "aside",
          "display1": "h1",
          "display2": "h1",
          "display3": "h1",
          "display4": "h1",
          "headline": "h1",
          "subheading": "h3",
          "title": "h2",
        }
      }
      noWrap={false}
      paragraph={false}
      style={
        Object {
          "color": "green",
        }
      }
      variant="subheading"
    >
      <h3
        className="MuiTypography-root-160 MuiTypography-subheading-167"
        style={
          Object {
            "color": "green",
          }
        }
      >
        Successfully installed! The Strand bot should have DM'd you a welcome message.
      </h3>
    </Typography>
  </WithStyles(Typography)>
</InstallationStatus>
`;
